x-pack/plugins/apm/public/locator/locator.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { merge } from 'lodash';
+import type { LocatorDefinition } from '@kbn/share-plugin/common';
+import { getLocatorEnabledRoutes } from './get_locator_enabled_routes';
+import type { APMLocatorPayload } from './types';
+
+export const APM_APP_LOCATOR_ID = 'APM_LOCATOR';
+
+async function loadApmRoutesModule() {
+  const { apmRouter, apmRoutes } = await import(
+    '../components/routing/apm_route_config'
+  );
+  const locatorEnabledRoutes = getLocatorEnabledRoutes(apmRoutes);
+
+  return {
+    apmRouter,
+    locatorEnabledRoutes,
+  };
+}
+
+export class APMLocatorDefinition
+  implements LocatorDefinition<APMLocatorPayload>
+{
+  readonly id = APM_APP_LOCATOR_ID;
+  apmRoutesModule = loadApmRoutesModule();
+
+  async getLocation(payload: APMLocatorPayload) {
```

Comment:
I might be misremembering, but I remember us having a discussion about whether to use the route definition to generate routes or not, and I thought the outcome of that discussion was that because it'd be better to have a simple function that only links to the service overview, because we don't get type safety (for consumers) anyway. I'm probably the one who suggested using the route definition, so apologies there, but I think this PR becomes a lot simpler if we just do this for the service overview route. We can still validate the route (runtime + typecheck). I'm not sure if we should go all in on locators, so would prefer to keep it simple here for this one use case.